core/tasks/campaigns/fire_campaign_event.go

```diff
@@ -231,3 +232,9 @@ func FireCampaignEvents(ctx context.Context, rt *runtime.Runtime, oa *models.Org
 
 	return handled, nil
 }
+
+func filterByWaitingSession(ctx context.Context, db *sqlx.DB, contacts []models.ContactID) ([]models.ContactID, error) {
```

Comment:
just moved this func into this file because it's only used by this task and we want to delete it when we delete the task